### ISSUE TITLE
About further charge at 100% SOC when fitting experiment data

### ISSUE THREAD

**Issue by tinosulzer** [pybamm-team/PyBaMM#2046]
### Discussed in https://github.com/pybamm-team/PyBaMM/discussions/2040

<div type='discussions-op-text'>

<sup>Originally posted by **RuiheLi** April 28, 2022</sup>
Hi PyBaMM,

Recently I am trying to fit the experiment data and find that error will occur when I try to charge the cell at 100% SOC. The code is is follows:

```
Model_9 = pb.lithium_ion.DFN()
chemistry = pb.parameter_sets.Chen2020
Para_0 = pb.ParameterValues(chemistry=chemistry)
V_max = 4.2;        V_min = 2.5;
Experiment_Breakin= pb.Experiment( [    # at 25 degreeC
    (f"Discharge at 0.2 C until {V_min} V",
    "Rest for 4 hours",
    f"Charge at 0.2 C until {V_max} V" , 
    f"Hold at {V_max} V until C/100",
    "Rest for 2 hours")        ]   *1
    )  
Experiment_RPT_Long    = pb.Experiment( [      # under 25 degreeC
    (
        f"Charge at 0.3 C until {V_max} V", 
        f"Hold at {V_max} V until C/100",
        "Rest for 2 hours",  
        f"Discharge at 0.1 C until {V_min} V",  
        "Rest for 6 hours",  
        f"Charge at 0.1 C until {V_max} V",
        "Rest for 10 minutes",  
    ) ] * 1 )
Sol_NonDry=[];  Model_NonDry=[];
Sim_0 = pb.Simulation(Model_9, experiment = Experiment_Breakin,parameter_values = Para_0,solver = pb.CasadiSolver(),)       
Sol_0 = Sim_0.solve();
Sol_NonDry.append(Sol_0); Model_NonDry.append(Model_9);

Model_1 = Model_9.set_initial_conditions_from(Sol_0, inplace=False)
Sim_1 = pb.Simulation(Model_1, experiment = Experiment_RPT_Long,parameter_values = Para_0,solver = pb.CasadiSolver(),)   
Sol_1 = Sim_1.solve(); 
Sol_NonDry.append(Sol_1); Model_NonDry.append(Model_1);
```

I understand it is a bit wired to do this in modelling. But that is quite normal in experiments. I know that if I delete the first two steps in Experiment_RPT_Long (further charge), the error will disappear. But that will make it more complicated to transfer the real test procedure into modelling steps. I wonder whether there are any easy way to solve this? Big thanks in advance!
</div>

**Comment by tinosulzer** [comment]
Another bug here https://twitter.com/battbot_/status/1547674287917518848

**Comment by tinosulzer** [comment]
fixed by #2212
